### PR TITLE
fix: attributes changed: compare timestamps with configurable threshold.

### PR DESCRIPTION
**Pull request by merlin-northern**
Changelog: Title
Ticket: MEN-6643

### CODE REVIEW

**Comment by merlin-northern** [comment]
@mender-test-bot start pipeline --fast please --pass please

File: config.yaml [hunk 1]
```diff
@@ -44,6 +44,13 @@ mongo: mongo-inventory:27017
 # Overwrite with environment variable: INVENTORY_LIMIT_ATTRIBUTES
 # limit_attributes: 100
 
+# Maximum number of seconds between the inventory updates, if
+# you updated the inventory more than this number of seconds it will
+# get updated unconditionally upon next request from a device
+# Defaults to: 86400
+# Overwrite with environment variable: INVENTORY_LAST_UPDATE_SECONDS_THRESHOLD
+# last_update_seconds_threshold: 259200
+
```
Comment (alfrunes) [inline config.yaml:53]: The default and the value differs.
Also, see :point_up: comment.

**Review by alfrunes — COMMENTED** [review]
Logic seems fine, I just have some UX and readability improvement requests (feel free to challenge them).

File: config.yaml (hunk 1 above)
Comment (merlin-northern) [reply config.yaml:53]: on purpose :> who says the example should follow the default :>
ok I will move to durations

File: config.go [hunk 2]
```diff
@@ -39,6 +39,9 @@ const (
 	SettingLimitAttributes        = "limit_attributes"
 	SettingLimitAttributesDefault = 100
 
+	SettingLastUpdateSecondsThreshold        = "last_update_seconds_threshold"
+	SettingLastUpdateSecondsThresholdDefault = 24 * 60 * 60
```
[the remark this answers is not in the document]
Comment (merlin-northern) [reply config.go:43]: sure

File: store/mongo/datastore_mongo.go [hunk 3]
```diff
@@ -372,16 +373,11 @@ func (db *DataStoreMongo) inventoryNeedsUpdate(
...
-			if now.Day() != lastUpdatedDate.Day() {
-				if now.Month() != lastUpdatedDate.Month() {
-					if now.Year() != lastUpdatedDate.Year() {
-						devicesInNeedOfUpdate = append(devicesInNeedOfUpdate, device.ID)
-						continue
-					}
-				}
+			lastUpdateTime := v.Time()
+			now := time.Now()
+			if (now.Unix() - lastUpdateTime.Unix()) > lastUpdateSecondsThreshold {
```
[the remark this answers is not in the document]
Comment (merlin-northern) [reply store/mongo/datastore_mongo.go:378]: this will not work as you expect, without explicit UTC()  on 376 and 377. ut no worries, I will adjust.

**Comment by merlin-northern** [comment]
> Logic seems fine, I just have some UX and readability improvement requests (feel free to challenge them).

I will just code them instead of challenging, works? :)

**Comment by merlin-northern** [comment]
@mender-test-bot start pipeline --fast --and --pass please

**Review by alfrunes — APPROVED** [review]

**Pull request merged**